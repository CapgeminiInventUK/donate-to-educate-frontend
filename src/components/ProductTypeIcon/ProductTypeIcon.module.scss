@use '@assets/global.scss' as g;

.productType {
  margin-right: 10px;
  display: inline-block;
  height: 30px;

  .innerSpan {
    display: inline-block;
    height: 30px;
  }
}
