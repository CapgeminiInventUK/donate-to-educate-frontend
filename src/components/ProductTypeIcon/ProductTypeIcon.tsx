import { FC } from 'react';
import styles from './ProductTypeIcon.module.scss';
import { convertNumberToCategory } from '../ItemList/getFullItemList';
import { Popover } from 'antd';
import ProductTypes from '@/assets/icons/ProductTypes';
import { ProductTypeIconProps } from '@/types/props';

const ProductTypeIcon: FC<ProductTypeIconProps> = ({ productType, colour }) => {
  return (
    <span key={productType} className={styles.productType}>
      <Popover content={convertNumberToCategory(productType)} trigger="hover">
        <span className={styles.innerSpan}>
          <ProductTypes type={productType} colour={colour} />
        </span>
      </Popover>
    </span>
  );
};
export default ProductTypeIcon;
