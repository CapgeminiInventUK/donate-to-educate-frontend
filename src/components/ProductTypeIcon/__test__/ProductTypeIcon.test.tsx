import { render } from '@testing-library/react';
import ProductTypeIcon from '../ProductTypeIcon';

describe('Product type icon', () => {
  it('should render component', () => {
    expect(render(<ProductTypeIcon colour="#00B6A8" productType={1} />)).toMatchSnapshot();
  });
});
