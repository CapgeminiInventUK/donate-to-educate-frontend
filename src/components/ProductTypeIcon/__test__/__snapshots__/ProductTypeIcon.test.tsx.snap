// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Product type icon > should render component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="_productType_d2b676"
      >
        <span
          class="_innerSpan_d2b676"
        >
          <svg
            fill="none"
            height="30"
            viewBox="0 0 30 30"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="#00B6A8"
              height="30"
              rx="15"
              width="30"
            />
            <path
              d="M15 23.3569C19.6155 23.3569 23.3571 19.6152 23.3571 14.9997C23.3571 10.3842 19.6155 6.64258 15 6.64258C10.3844 6.64258 6.64282 10.3842 6.64282 14.9997C6.64282 19.6152 10.3844 23.3569 15 23.3569Z"
              fill="#FEFCFD"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
            <path
              d="M15 6.64258V23.3569"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
            <path
              d="M8.69995 20.4898C9.64204 19.9162 10.4207 19.11 10.9611 18.1485C11.5015 17.187 11.7855 16.1027 11.7857 14.9998C11.7855 13.8968 11.5015 12.8125 10.9611 11.851C10.4207 10.8896 9.64204 10.0833 8.69995 9.50977"
              fill="#FEFCFD"
            />
            <path
              d="M8.69995 20.4898C9.64204 19.9162 10.4207 19.11 10.9611 18.1485C11.5015 17.187 11.7855 16.1027 11.7857 14.9998C11.7855 13.8968 11.5015 12.8125 10.9611 11.851C10.4207 10.8896 9.64204 10.0833 8.69995 9.50977"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
            <path
              d="M21.3 9.50977C20.3583 10.0835 19.5799 10.8899 19.0398 11.8513C18.4997 12.8128 18.2161 13.897 18.2161 14.9998C18.2161 16.1025 18.4997 17.1868 19.0398 18.1482C19.5799 19.1097 20.3583 19.916 21.3 20.4898"
              fill="#FEFCFD"
            />
            <path
              d="M21.3 9.50977C20.3583 10.0835 19.5799 10.8899 19.0398 11.8513C18.4997 12.8128 18.2161 13.897 18.2161 14.9998C18.2161 16.1025 18.4997 17.1868 19.0398 18.1482C19.5799 19.1097 20.3583 19.916 21.3 20.4898"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </span>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="_productType_d2b676"
    >
      <span
        class="_innerSpan_d2b676"
      >
        <svg
          fill="none"
          height="30"
          viewBox="0 0 30 30"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="#00B6A8"
            height="30"
            rx="15"
            width="30"
          />
          <path
            d="M15 23.3569C19.6155 23.3569 23.3571 19.6152 23.3571 14.9997C23.3571 10.3842 19.6155 6.64258 15 6.64258C10.3844 6.64258 6.64282 10.3842 6.64282 14.9997C6.64282 19.6152 10.3844 23.3569 15 23.3569Z"
            fill="#FEFCFD"
            stroke="#000000"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M15 6.64258V23.3569"
            stroke="#000000"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M8.69995 20.4898C9.64204 19.9162 10.4207 19.11 10.9611 18.1485C11.5015 17.187 11.7855 16.1027 11.7857 14.9998C11.7855 13.8968 11.5015 12.8125 10.9611 11.851C10.4207 10.8896 9.64204 10.0833 8.69995 9.50977"
            fill="#FEFCFD"
          />
          <path
            d="M8.69995 20.4898C9.64204 19.9162 10.4207 19.11 10.9611 18.1485C11.5015 17.187 11.7855 16.1027 11.7857 14.9998C11.7855 13.8968 11.5015 12.8125 10.9611 11.851C10.4207 10.8896 9.64204 10.0833 8.69995 9.50977"
            stroke="#000000"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M21.3 9.50977C20.3583 10.0835 19.5799 10.8899 19.0398 11.8513C18.4997 12.8128 18.2161 13.897 18.2161 14.9998C18.2161 16.1025 18.4997 17.1868 19.0398 18.1482C19.5799 19.1097 20.3583 19.916 21.3 20.4898"
            fill="#FEFCFD"
          />
          <path
            d="M21.3 9.50977C20.3583 10.0835 19.5799 10.8899 19.0398 11.8513C18.4997 12.8128 18.2161 13.897 18.2161 14.9998C18.2161 16.1025 18.4997 17.1868 19.0398 18.1482C19.5799 19.1097 20.3583 19.916 21.3 20.4898"
            stroke="#000000"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </span>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
